src/virtual-source-buffer.js

```diff
@@ -247,29 +240,33 @@ export default class VirtualSourceBuffer extends videojs.EventTarget {
     let types = ['audio', 'video'];
 
     types.forEach((type) => {
+      // Don't create a SourceBuffer of this type if we don't have a
+      // codec for it
       if (!this[`${type}Codec_`]) {
         return;
       }
 
-      // if the mediasource already has a buffer for the codec
-      // use that
-      if (this.mediaSource_[`${type}Buffer_`]) {
-        this[`${type}Buffer_`] = this.mediaSource_[`${type}Buffer_`];
+      // Do nothing if a SourceBuffer of this type already exists
+      if (this[`${type}Buffer_`]) {
```

Comment:
I think we haven't properly setup our object-ownership hierarchies if we find ourselves regularly interactive with sub-sub-properties but am ok with coming back to this later. Thank you for the rename.